### PR TITLE
Resource leak checker documentation

### PR DESCRIPTION
**Pull request by kelloggm**

### CODE REVIEW

File: docs/manual/resource-leak-checker.tex [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+A post-condition annotation,
+\<@CreatesObligation>,
+indicates for which expression an obligation is created.
+
+When checking a call to a method annotated as
+\<@CreatesObligation(>$expr$\<)>, the Resource Leak Checker
+(1) treats the \<@MustCall>
+obligation of $expr$ as \emph{satisfied},
+and (2) creates a fresh obligation to check.
+This is sound: the checker creates a new obligation for calls to
```
Comment (mernst) [inline docs/manual/resource-leak-checker.tex:236]: Please replace indefinite references with specific ones.  I'm not sure whether "this" refers to both 1&2 or just 2.  I have similar questions about "this check" and "the latter".

**Review by mernst — CHANGES_REQUESTED** [review]

File: docs/manual/resource-leak-checker.tex [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+In a class that has must-call obligations,
+every constructor implicitly creates obligations for the newly-created object.
+However, non-constructor methods may also create obligations
+when re-assigning non-final, owning fields or allocating
+new system-level resources.
+A post-condition annotation,
+\<@CreatesObligation>,
+indicates for which expression an obligation is created.
+
+When checking a call to a method annotated as
```
[the remark this answers is not in the document]
Comment (kelloggm) [reply docs/manual/resource-leak-checker.tex:231]: The only requirements on a method with an `@CreatesObligation` annotation are:
* the target is in-scope and parseable
* the method does not override a method with a weaker `@CreatesObligation` annotation (including no such annotation)

In general, these requirements are pretty easy to fulfill. The requirement you're referring to in your question doesn't come from the `@CreatesObligation` annotation itself, but rather from re-assigning a field (which also creates a requirement to write an `@CreatesObligation` annotation, which I think explains your confusion).

I added a note to the text that explains those two conditions on where the annotation can be written, and extended the discussion generally. Let me know if this is still not clear.

**Comment by kelloggm** [comment]
@mernst, I've addressed your PR comments. Can you take another look? Thanks!

**Comment by kelloggm** [comment]
I've addressed the comments on this PR in the `resource-leak-checker-full` branch, which will be the basis for a PR to the CF proper. I'm closing this PR now; further review will take place on that future PR.

**Pull request closed without merging**